### ISSUE TITLE
Use of '*' in API queries to get all results .

### ISSUE THREAD

**Issue by sagarpreet-chadha** [publiclab/plots2#4036]
We have this API implemented here : 

https://github.com/publiclab/plots2/blob/ba053502ded9ec0163af90d3d7a4cb3eb2a4b74e/app/api/srch/search.rb#L224

We need to fetch all the results (they can be thousands in number!)  if the query parameter used is 'question:*' . 

For example : 
`/api/srch/taglocations?tagName=question:*`

**Comment by oorjitchowdhary** [comment]
@sagarpreet-chadha Could you please explain me this issue..

**Comment by sagarpreet-chadha** [comment]
Hi sure 😄 !
We basically want to get all the question-posts on publiclab.org which have latitude and longitude tags to be returned from the above API . 
Makes sense ? Thanks !

**Comment by oorjitchowdhary** [comment]
Apologies if I'm wrong but is it similar to this issue.. https://github.com/publiclab/leaflet-environmental-layers/issues/30

**Comment by eel87** [comment]
@sagarpreet-chadha  Is this issue still open? If so I would like to try and help!

**Comment by sagarpreet-chadha** [comment]
Yes i think it is still open. @cesswairimu can you please verify? Thanks!

**Comment by KarishmaVanwari** [comment]
Can I take up this issue, if it is open? @sagarpreet-chadha

**Comment by Pragati1419** [comment]
I would love to continue with this issue. Can you assign it to me if it is still open or not assigned to anyone else?

**Comment by bad-atnames0** [comment]
Hi @sagarpreet-chadha ,I would love to continue with this issue. Can you assign it to me if it is still open or not assigned to anyone else?

**Comment by Laraibnehal** [comment]
@sagarpreet-chadha  Can you explain me this issue I wanna work on it please assign it to me.

**Comment by itsgurmannatsohal** [comment]
Is this issue open? Can I take this up?

**Comment by jywarren** [comment]
Hello, to be honest I think this is a tough one to implement, and perhaps we would prefer to put a hold on it for now. Thank you very much for your offers of help!

**Comment by Anubhab2003** [comment]
Please assign me this

**Comment by Prasad-JB** [comment]
Hi, I’m new to the project and would like to work on this issue as my first contribution. Could you please assign it to me?

**Comment by Prasad-JB** [comment]
Hi, I’m interested in working on this issue. I’ve read through the problem description and the relevant file (plots2/app/api/srch/search.rb).
Could you please assign it to me so I can start working on a solution?

Thank you!

**Comment by rajshekharbind** [comment]
Hello, to be honest I think this is a tough one to implement, and also learnt  postman api  and also some work we would prefer to put a hold on it for now.

**Comment by JanviBhateja** [comment]
Hi, I’m Janvi, a 2nd year ENC student and new to contributing to Public Lab.
I wanted to ask if this issue is still relevant and suitable for a first-time contributor.
I’d be happy to work on a smaller scoped task if you suggest one.
Thank you!

**Comment by afifakhan123** [comment]
Hi! I’d like to work on this issue.
I’m exploring the logic around tagName == "question:*" in search.rb
and will propose a small conditional fix to fetch all results.
Please let me know if that sounds good. Thanks!

**Comment by afifakhan123** [comment]
I’ve added a conditional override so that when tagName=question:* is used,
the API fetches all matching tag locations by increasing the rows limit.
Happy to adjust if a different approach is preferred. Thanks!

**Comment by ShauryaCodes2004** [comment]
Hi! I’d like to work on this issue if it’s still open.
Please let me know if I can take it. Thanks!

**Comment by afifakhan123** [comment]
Hi! I’d like to work on this issue. I’m currently exploring the API search behavior in plots2 and would love to help improve support for * queries. Please assign this to me if available. Thanks!

**Comment by 786imran786** [comment]
Hi, I’d like to work on this issue. Could you please guide me if this is still open?

**Comment by amisha1072006** [comment]
Hi! @sagarpreet-chadha , I’d like to work on this issue if it’s still available.
Please let me know if I can proceed. Thanks!

**Comment by sanithrshetty15** [comment]
Hi, I’d like to work on this issue.
I believe the default `queryLimit` of 5 is causing incomplete results for `question:*`.
My plan is to skip applying the limit only when `tagName === "question:*"`.
Please let me know if this approach is okay. Thanks!

**Comment by kaashvi5** [comment]
Hi, I’m a beginner and I’d like to work on this issue. Can you please guide me?

**Comment by Sudarshan-Raut** [comment]
Hi, I am a beginner and I would like to work on this issue. Can you please assign it to me or guide me?
